src/server/qgsserverapiutils.cpp

```diff
@@ -61,6 +62,380 @@ QgsRectangle QgsServerApiUtils::parseBbox( const QString &bbox )
   return QgsRectangle();
 }
 
+QList< QgsVectorLayerServerProperties::WmsDimensionInfo > QgsServerApiUtils::temporalDimensions( const QgsVectorLayer *layer )
+{
+  QList< QgsVectorLayerServerProperties::WmsDimensionInfo > dimensions { layer->serverProperties()->wmsDimensions() };
+  // Filter only date and time
+  dimensions.erase( std::remove_if( dimensions.begin(),
+                                    dimensions.end(),
+                                    [ ]( QgsVectorLayerServerProperties::WmsDimensionInfo & dim )
+  {
+    return dim.name.toLower() != QStringLiteral( "time" )
+           && dim.name.toLower() != QStringLiteral( "date" ) ;
+  } ), dimensions.end() );
+
+  // Automatically pick up the first date/datetime field if dimensions is empty
```

Comment:
This is just temporary logic, right? It would be removed when work from https://github.com/qgis/QGIS-Enhancement-Proposals/issues/128 begins to land?

Reply:
Yes, I think so. What might be eventually reused and moved to core (QgsExpression?) is part of https://github.com/qgis/QGIS/pull/32322/files/b6e6dccf02b0684b2410cf41919e6c6f3b06781b#diff-7d29a5fc7ae368ec22273c10d5c2a0e7R139